cartridge-cli.lua

```diff
@@ -1651,7 +1676,240 @@ local function run_hook(dir, filename)
+
+    -- Pass DOWNLOAD_TOKEN build arg for Tarantool Enterprise
+    if app_state.tarantool_is_enterprise then
+        local download_token_arg = string.format(
+            '--build-arg DOWNLOAD_TOKEN=%s',
+            app_state.download_token
+        )
+        table.insert(docker_build_args, download_token_arg)
+    end
+
+    -- Use base image as a cache
+    local cache_from_base_arg = string.format(
+        '--cache-from %s',
+        app_state.base_image_fullname
+    )
+    table.insert(docker_build_args, cache_from_base_arg)
+
+    return table.concat(docker_build_args, ' ')
+end
+
+-- * ---------------------- Building application itself ----------------------
+
+local function copy_tarantool_binaries(dir)
+    info('Copy Tarantool Enterprise binaries')
+    assert(app_state.tarantool_is_enterprise)
+
+    local tarantool_dir = get_tarantool_dir()
+
+    for _, binary in ipairs({'tarantool', 'tarantoolctl'}) do
+        local path_from = fio.pathjoin(tarantool_dir, binary)
+        local path_to = fio.pathjoin(dir, binary)
+
+        local ok, err = copyfile(path_from, path_to)
+        if not ok then return false, err end
+    end
+
+    return true
+end
+
+local function build_application_in_docker(dir)
+    assert(app_state.build_in_docker)
+    -- Application is built in docker container:
+    -- - First, the base docker image <app-name>-base is built.
+    -- - Then, build commands are run on this image
+    --   with volume in the local dir
+    -- As a result, we have application with rocks modules
+    --   specific for a target platform in the local dir
+
+    local docker = which('docker')
+    if docker == nil then
+        return false, 'docker binary is required to build application in docker'
+    end
+
+    -- Build the base image
+    info('Building docker image: %s', app_state.base_image_fullname)
+
+    -- - Write base image Dockerfile
+    local base_dockerfile_path = fio.pathjoin(fio.tempdir(), 'Dockerfile.cartridge.base')
+    local base_dockerfile_content, err = construct_build_image_dockerfile()
+    if base_dockerfile_content == nil then return false, err end
+
+    local ok, err = write_file(base_dockerfile_path, base_dockerfile_content)
+    if not ok then return false, err end
+
+    -- - Build the base docker image
+    local build_image_command = expand(BUILD_IMAGE_COMMAND_TEMPLATE, {
+        docker = docker,
+        distribution_dir = dir,
+        image_fullname = app_state.base_image_fullname,
+        dockerfile_path = base_dockerfile_path,
+        docker_build_args = get_docker_build_args_string(),
+    })
+    local ok, err = call(build_image_command)
+    if not ok then
+        return false, string.format('Failed to build image: %s', err)
+    end
+
+    info('Base image tagged as: %s', app_state.base_image_fullname)
+
+    -- Build application in the base image
+    info('Build application in %s', app_state.base_image_fullname)
```

Comment:
I think it's a copypaste error because of `app_state.base_image_fullname` in both places

Reply:
Yes, but I'm not sure that it's bad to show both these messages.